### PR TITLE
test: add a test that have over 2 times of limit files per plan

### PR DESCRIPTION
**Pull request by NGA-TRAN**
@alamb This is the test you request

Connects to https://github.com/influxdata/influxdb_iox/pull/7007 and https://github.com/influxdata/idpe/issues/17090

### CODE REVIEW

File: ingester2/tests/common/mod.rs [hunk 1]
```diff
@@ -375,7 +375,7 @@ where
 
         self.metrics
             .get_instrument::<Metric<U>>(name)
-            .unwrap_or_else(|| panic!("failed to find metric {}", name))
+            .unwrap_or_else(|| panic!("failed to find metric {name}"))
```
Comment (NGA-TRAN) [inline ingester2/tests/common/mod.rs:378]: clippy requests me to do so

Comment (alamb) [reply ingester2/tests/common/mod.rs:378]: Yeah, I saw that locally as well

File: compactor2/tests/layouts/mod.rs [hunk 2]
```diff
@@ -1347,6 +1347,877 @@ async fn many_tiny_l0_files() {
...
+    - "L0.402[401,402]             |L0.402|                                                                "
+    - "L0.403[402,403]                     |L0.403|                                                        "
+    - "L0.404[403,404]                             |L0.404|                                                "
+    - "L0.405[404,405]                                     |L0.405|                                        "
+    - "L0.406[405,406]                                             |L0.406|                                "
+    - "L0.407[406,407]                                                     |L0.407|                        "
+    - "L0.408[407,408]                                                             |L0.408|                "
+    - "L0.409[408,409]                                                                     |L0.409|        "
+    - "L0.410[409,410]                                                                             |L0.410|"
+    - "**** Simulation run 3, type=compact. 3 Input Files, 2.8mb total:"
```
Comment (alamb) [inline compactor2/tests/layouts/mod.rs:2209]: 👍 looks great

**Review by alamb — APPROVED** [review]
Thank you @NGA-TRAN

**Pull request merged**